### PR TITLE
updated to 1.20.0

### PR DESCRIPTION
**Pull request by ValerasNarbutas**
> By submitting this pull request, you agree to the [contribution guidelines](https://github.com/pnp/sp-dev-fx-webparts/blob/main/CONTRIBUTING.md)

> If you aren't familiar with how to contribute to open-source repositories using GitHub, or if you find the instructions on this page confusing, [sign up](https://forms.office.com/Pages/ResponsePage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUREZVRDVYUUJLT1VNRDM4SjhGMlpUNzBORy4u) for one of our [Sharing is Caring](https://pnp.github.io/sharing-is-caring/#pnp-sic-events) events. It's completely free, and we'll guide you through the process.

> To submit a pull request with multiple authors, make sure that at least one commit is a co-authored commit by adding a `Co-authored-by:` trailer to the commit's message. E.g.: `Co-authored-by: name <name@example.com>`

> Put an `x` in all the items that apply (`[x]`, no spaces between the `[`, the `x`, and the `]` ), make notes next to any that haven't been addressed.

- [ ] New sample
- [x] Bug fix/update
- [x] Related issues: #5252

## What's in this Pull Request?

SPFX upgrade to 1.20.0 version

## Node Version
Node version used: 18.17.1

## Checklist

- [x] My pull request affects only ONE sample.
- [x] My sample builds without any warnings
- [x] I have updated the `README.md` file's **Version history**. For new samples, created a new `README.md` file matching [this template](templates/README-template.md)
- [x] My `README.md` has at least one static high-resolution screenshot (i.e. not a GIF) located in the `assets` folder.
- [x] My `README.md` contains complete setup instructions, including pre-requisites and permissions required
- [x] My solution includes a `.nvmrc` file indicating the version of Node.js

### CODE REVIEW

**Comment by Adam-it** [comment]
Thank you for opening a PR to our samples repo and helping us upgrade our project 👍
We will try to review this PR ASAP

File: samples/react-birthdays/src/assets/sample.json
Comment (Adam-it) [inline samples/react-birthdays/src/assets/sample.json:1]: I think this file should not be here. The `sample.json` should be present in the `assets` folder in the root of the project and is used to store metadata for sample galleries. We should remove this instance here and leave only the one we already have in the `react-birthday/assets`
Along the way there we should update the `SPFX-VERSION` to 1.20.0

File: samples/react-birthdays/package.json [hunk 1]
```diff
@@ -1,58 +1,47 @@
 {
-  "name": "birthdays-anniversary",
-  "version": "2.0.0",
+  "name": "happy-birthday",
```
Comment (Adam-it) [inline samples/react-birthdays/package.json:2]: why change the name?

File: samples/react-birthdays/package.json [hunk 2]
```diff
@@ -1,58 +1,47 @@
...
+  "version": "0.0.1",
```
Comment (Adam-it) [inline samples/react-birthdays/package.json:3]: we should  either not change the version number to make it higher but not smaller

**Review by Adam-it — CHANGES_REQUESTED** [review]
@ValerasNarbutas Awesome work 👏👏👏 You Rock 🤩
I managed to build it locally and played with it a bit.
During my review I noticed some small details we could sort out before we merge 👍. Please do give them a double check 🙏

**Comment by hugoabernier** [comment]
@ValerasNarbutas do you mind taking a look at the code or instructions. The build seems to produce the following error in certain cases:
{
    "statusCode": 400,
    "code": "invalidRequest",
    "requestId": "3dbfa519-574b-41b2-b9e2-42b605801c77",
    "date": "2024-10-12T20:43:53.000Z",
    "body": "{\"code\":\"invalidRequest\",\"message\":\"Field 'Birthday' cannot be referenced in filter or orderby as it is not indexed. Provide the 'Prefer: HonorNonIndexedQueriesWarningMayFailRandomly' header to allow this, but be warned that such queries may fail on large lists.\",\"innerError\":{\"date\":\"2024-10-12T16:43:53\",\"request-id\":\"3dbfa519-574b-41b2-b9e2-42b605801c77\",\"client-request-id\":\"429059bb-c957-5969-ae5a-84515e5d9870\"}}"
}

**Comment by ValerasNarbutas** [comment]
> @ValerasNarbutas do you mind taking a look at the code or instructions. The build seems to produce the following error in certain cases:
> {
>     "statusCode": 400,
>     "code": "invalidRequest",
>     "requestId": "3dbfa519-574b-41b2-b9e2-42b605801c77",
>     "date": "2024-10-12T20:43:53.000Z",
>     "body": "{\"code\":\"invalidRequest\",\"message\":\"Field 'Birthday' cannot be referenced in filter or orderby as it is not indexed. Provide the 'Prefer: HonorNonIndexedQueriesWarningMayFailRandomly' header to allow this, but be warned that such queries may fail on large lists.\",\"innerError\":{\"date\":\"2024-10-12T16:43:53\",\"request-id\":\"3dbfa519-574b-41b2-b9e2-42b605801c77\",\"client-request-id\":\"429059bb-c957-5969-ae5a-84515e5d9870\"}}"
> }

Will do @hugoabernier

**Comment by ValerasNarbutas** [comment]
thanks for checking  @Adam-it  and @hugoabernier , will review this.

**Comment by ValerasNarbutas** [comment]
> Birthday

Hey @hugoabernier , did you indexed "Birthday", it is in instructions
image

**Comment by hugoabernier** [comment]
Many thanks @ValerasNarbutas for your update! Awesome!.

Thank you for sharing your sample with others - you rock! 👏🥇👩‍💻

**Pull request merged**